cpp/include/raft/linalg/detail/map.cuh

```diff
@@ -116,15 +116,28 @@ struct ratio_selector {
   template <typename T>
   constexpr static auto ignoring_alignment() -> ratio_selector
   {
-    return ratio_selector{raft::div_rounding_up_safe<size_t>(kCoalescedVectorSize, sizeof(T)), 0};
+    constexpr bool T_evenly_fits_in_cache_line = (kCoalescedVectorSize % sizeof(T)) == 0;
+
+    if constexpr (T_evenly_fits_in_cache_line) {
+      return ratio_selector{size_t(kCoalescedVectorSize / sizeof(T)), 0};
+    } else {
+      return ratio_selector{1, 0};
+    }
   }
 
   template <typename T>
   explicit ratio_selector(const T* ptr)
   {
     constexpr auto s = ignoring_alignment<T>();  // NOLINT
-    align            = int(Pow2<sizeof(T) * s.ratio>::roundUp(ptr) - ptr);
-    ratio            = int(s.ratio);
+
+    constexpr bool T_evenly_fits_in_cache_line = (kCoalescedVectorSize % sizeof(T)) == 0;
+
+    if constexpr (T_evenly_fits_in_cache_line) {
+      align = int(Pow2<sizeof(T) * s.ratio>::roundUp(ptr) - ptr);
+    } else {
+      align = 0;
+    }
+    ratio = int(s.ratio);
```

Comment:
After discussion: 

The if else statement was changed to depend on `s.ratio` to better express intent. The align is still set to 0 directly in one of the branches (rather than using the roundup machinery) because Pow2::roundup does not support sizes that aren't powers of two.